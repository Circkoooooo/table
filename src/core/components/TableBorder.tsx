import React from "react"
import styled, { css } from "styled-components"

interface BorderWrapperProps {
	width: number
	height: number
	offsetLeft: number
	offsetTop: number
}

const BorderWrapper = styled.div<BorderWrapperProps>(({ width, height, offsetLeft, offsetTop }) => {
	return css`
		border: 2px solid red;
		width: ${width}px;
		height: ${height}px;
		position: absolute;
		z-index: 3;
		pointer-events: none;
		transform: translate(${offsetLeft}px, ${offsetTop}px);
	`
})

interface BaseBorderProps {
	isRender: true
	borderWidth: number
	borderHeight: number
	offsetLeft: number
	offsetTop: number
}

export type BorderProps =
	| ({
			isRender: false
	  } & Omit<Partial<BaseBorderProps>, "isRender">)
	| BaseBorderProps

const TableBorder: React.FC<BorderProps> = ({ isRender, borderWidth, borderHeight, offsetLeft, offsetTop }) => {
	if (!isRender) return null

	return <BorderWrapper height={borderHeight} width={borderWidth} offsetLeft={offsetLeft} offsetTop={offsetTop} />
}

export default TableBorder
