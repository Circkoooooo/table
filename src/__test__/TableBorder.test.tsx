import { render } from "@testing-library/react"
import TableBorder, { BorderWrapper } from "../core/components/TableBorder"

describe("Table border component", () => {
	it("border wrap component", () => {
		const { asFragment } = render(<BorderWrapper width={100} height={100} $offsetLeft={10} $offsetTop={10} />)
		expect(asFragment()).toMatchSnapshot()
	})
})
