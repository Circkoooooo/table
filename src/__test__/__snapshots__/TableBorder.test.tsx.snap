// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table border component border wrap component 1`] = `
<DocumentFragment>
  <div
    class="sc-bcPKhP kAXhXo"
    height="100"
    offsetleft="10"
    offsettop="10"
    width="100"
  />
</DocumentFragment>
`;

exports[`Table border component table border render 1`] = `
<styled.div
  height={100}
  offsetLeft={0}
  offsetTop={0}
  width={100}
/>
`;
